lib/internal/process/warning.js

```diff
@@ -46,4 +57,8 @@ function setupProcessWarnings() {
     }
     process.nextTick(() => process.emit('warning', warning));
   };
+
+  process.emitExperimentalWarning = function(name) {
+    process.emitWarning(new ExperimentalWarning(name));
```

Comment:
There's no reason to create a separate class for this, something like the following would work easily...

```js
const msg = `${feature} is an experimental feature. ` +
            'Implementation details may change.';
process.emitWarning(msg, 'ExperimentalWarning');
```